### PR TITLE
Reinstate Text control outside of settings in Link Control

### LINKED ISSUE #50948: Move `Text` field back into main interface of Link Control
Part of https://github.com/WordPress/gutenberg/issues/50890.

Let's revert the change that saw the `Text` field of Link Control moved into the settings drawer and instead reinstate it next to the `URL` field.

**Condition**: it should _only_ appear once the user has created the initial link (i.e. there is a value for `url` in the link `value` object.

### PR DESCRIPTION
**Pull request by getdave**
## What?

Reinstates `Text` input control outside of the settings drawer area of Link Control.

Part of https://github.com/WordPress/gutenberg/issues/50891

Closes https://github.com/WordPress/gutenberg/issues/50948

## Why?

Because the UX requires it.

## How?

Move the control outside the drawer and remove drilled props.

## Testing Instructions

- Create link and see no Text field
- Reselect link and edit link
- See text field
- Trying changing text and ensure it syncs up
- Do same on Nav block and Paragraph
Screenshot 2023-05-25 at 13 32 22

### CODE REVIEW

**Comment by getdave** [comment]
This PR has also allowed me to notice a potential refactor of settings drawer which I will do in a follow up.

Need to also fix the tests here.

**Review by scruffian — APPROVED** [review]
LGTM

**Review by richtabor — APPROVED** [review]

**Comment by getdave** [comment]
We are looking to migrate the Puppeteer tests to Playwright for `Links`. Once done we can rebase this (and other PRs) which will allow us to move forward more quickly.

**Comment by getdave** [comment]
@MaggieCabrera Once your test refactor PRs are merged I think it would be good to rebase here and try and get this in as soon as possible.

**Comment by getdave** [comment]
Currently failing on

```
const waitForURLFieldAutoFocus = async () => {
    > 20 | 		await page.waitForFunction(
         | 		           ^
      21 | 			() => !! document.activeElement.closest( '.block-editor-url-input' )
      22 | 		);
      23 | 	};
```

**Comment by MaggieCabrera** [comment]
I made a fix for the tests because when we introduce the new input, the modal puts the focus on the text input when opened instead of the URL. I wonder if that's the behavior we want. In any case, the fixes for the e2e are merely a bandaid since these tests are incredibly dependent on the UI, and tabbing to get to where you want. I'll be working on migrating these all to playwright and rewriting them so they are more UI agnostic unless the purpose of the test is to focus on keyboard navigation.

**Comment by MaggieCabrera** [comment]
@draganescu looking at [the comments](https://github.com/WordPress/gutenberg/blob/a2c9677853470ef13221d721a87c0fcc823ad04b/packages/block-editor/src/components/link-control/index.js#L184) on the code that handles focus here I'm not so sure if we want the URL to be focused here (it was the text input before we removed it). In any case, maybe that can be handled in a follow-up?

The tests pass now, so this can be merged, I'm not sure if the focus issue is worth blocking this one. Happy to hear anyone else's thoughts.

**Comment by richtabor** [comment]
> The tests pass now, so this can be merged, I'm not sure if the focus issue is worth blocking this one. 

Yea I think it's fine to merge. The edit view is getting revisited as a whole.

**Pull request merged**
